ledger-service/http-json-perf/src/main/scala/com/daml/http/perf/Main.scala

```diff
@@ -116,51 +120,81 @@ object Main extends StrictLogging {
     logger.info(s"$config")
 
     val et: ET[ExitCode] = for {
-      ledgerId <- either(
-        getLedgerId(config.jwt).leftMap(_ =>
-          new IllegalArgumentException("Cannot infer Ledger ID from JWT")
-        )
-      ): ET[LedgerId]
 
       _ <- either(
         config.traverse(s => resolveSimulationClass(s))
       ): ET[Config[Class[_ <: Simulation]]]
 
       exitCode <- rightT(
-        main2(ledgerId, config)
+        main2(config)
       ): ET[ExitCode]
 
     } yield exitCode
 
     et.run
   }
 
-  private def main2(ledgerId: LedgerId, config: Config[String])(implicit
+  private def main2(config: Config[String])(implicit
       asys: ActorSystem,
       mat: Materializer,
       aesf: ExecutionSequencerFactory,
       ec: ExecutionContext,
       elg: EventLoopGroup,
   ): Future[ExitCode] =
-    withLedger(config.dars, ledgerId.unwrap) { (ledgerPort, _, _) =>
-      withJsonApiJdbcConfig(config.queryStoreIndex) { jsonApiJdbcConfig =>
-        withHttpService(
-          ledgerId.unwrap,
-          ledgerPort,
-          jsonApiJdbcConfig,
-          None,
-        ) { (uri, _, _, _) =>
-          runGatlingScenario(config, uri.authority.host.address, uri.authority.port)
-            .flatMap { case (exitCode, dir) =>
-              toFuture(generateReport(dir))
-                .map { _ =>
-                  logger.info(s"Report directory: ${dir.toAbsolutePath}")
-                  exitCode
-                }
-            }: Future[ExitCode]
+    for {
+      // For custom tokens we need to extract the expected ledger id, however for
+      // user tokens we can just choose any ledger id, because it doesn't contain any.
+      // If we have a user token we also extract it here so we can allocate the user later
+      // after we started the ledger.
+      userAndLedgerId <- HttpService
+        .decodeJwt(config.jwt)
+        .leftMap(_.message)
+        .flatMap { decodedJwt =>
+          CreateFromUserToken
+            .parseAndDecodeUserToken(decodedJwt)
+            .bimap(
+              error => s"This is neither a custom token nor a user token: ${error.message}",
+              {
+                case StandardJWTPayload(userId, _, _) =>
+                  (Some(userId), LedgerId("perf-runner"))
+                case CustomDamlJWTPayload(ledgerIdPayload, _, _, _, _, _, _) =>
+                  (None, LedgerId(ledgerIdPayload.getOrElse("perf-runner")))
```

Comment:
```suggestion
                case payload: StandardJWTPayload =>
                  (Some(payload.userId), LedgerId("perf-runner"))
                case payload: CustomDamlJWTPayload =>
                  (None, LedgerId(payload.ledgerId.getOrElse("perf-runner")))
```
Minor: the main advantage of using this approach is that if the `case class` adds or removes fields there won't be any change necessary here.